core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -140,7 +155,9 @@ public void write(T row) throws IOException {
       PathOffset previous = insertedRowMap.put(copiedKey, pathOffset);
       if (previous != null) {
         // TODO attach the previous row if has a positional-delete row schema in appender factory.
-        posDeleteWriter.delete(previous.path, previous.rowOffset, null);
+        PositionDelete<T> delete = PositionDelete.create();
```

Comment:
nit: extract these 3 lines to a method?

not directly related to this PR. wondering if the TODO comment is still applicable?

Reply:
Extracted the method.

I think we don't want to add to store the previous row to save memory. So we can't write it to the positional delete, but I didn't want to change an unrelated comment